### DISCUSSION TITLE
Interactive Transactions and Typescript

### DISCUSSION THREAD

**Discussion by khussell** [prisma/prisma#8700] [Q&A] [answered]
Hello all,

I am having trouble with trying to use the new interactive transaction feature. I am getting a typescript error with my argument type, it is expecting an array. Any ideas of what I can do? I am using version 2.29. I have run introspect and generate, still getting an error. I tried to return an array of documents with my callback, but still same error. Any ideas what I can do? I would be super appreciative! Thank you!

<img width="638" alt="Screen Shot 2021-08-11 at 3 07 23 PM" src="https://user-images.githubusercontent.com/45740230/129088372-fc6955fc-d838-427b-8751-97c86c3567dd.png">

<img width="533" alt="Screen Shot 2021-08-11 at 2 59 12 PM" src="https://user-images.githubusercontent.com/45740230/129087326-58ff3990-4d51-4279-9a03-d2c0b3d20f49.png">

Here is the typescript error:

<img width="967" alt="Screen Shot 2021-08-11 at 3 00 54 PM" src="https://user-images.githubusercontent.com/45740230/129087553-bd0ae050-7e08-42c3-8939-e713da9b330c.png">

**Comment by ryands17** [comment, marked as answer]
@khussell 👋 

You need to return something from the transaction. If you don't want to, then just a `return []` should work.

**Reply by khussell** [reply]
Huh, today I opened up vscode and the error went away with no changes... I guess I just needed to restart vscode 😩😂. Thank you for your help!

**Comment by Rykuno** [comment]
I'm still running into this issue. Restarting/reindexing does not work. 
```
    "@prisma/client": "2.29.1",
    "prisma": "2.29.1",
```


![Screen Shot 2021-08-14 at 7 34 45 PM](https://user-images.githubusercontent.com/10355852/129463428-bfe62560-69a5-4d1b-aa61-b4eff54b86d6.png)

Even with my operations essentially emulating the

**Comment by hugo-clemente** [comment]
This error is most likely linked with this already resolved issue: https://github.com/prisma/prisma/issues/8761#issuecomment-899477091

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
